woodwork/table_accessor.py

```diff
@@ -1247,7 +1248,9 @@ def _infer_missing_logical_types(
         parsed_logical_types[name] = _get_column_logical_type(
             series, logical_type, name
         )
-        updated_series = parsed_logical_types[name].transform(series)
+        parsed_logical_type = parsed_logical_types[name]
+        updated_series = parsed_logical_type.transform(series)
+        parsed_logical_type.validate(updated_series)
```

Comment:
I don't think we want to call this here. Validation is only when the user defines/sets a specific column to be a Logical Type.

Reply:
makes sense -- will remove